src/sidebar/services/load-annotations.js

```diff
@@ -16,53 +24,59 @@ export default function loadAnnotationsService(
   let searchClient = null;
 
   /**
-   * Load annotations for all URIs and groupId.
+   * Load annotations
    *
-   * @param {string[]} uris
-   * @param {string} groupId
+   * @param {LoadAnnotationOptions} options
    */
-  function load(uris, groupId) {
+  function load(options) {
+    const { groupId, uris } = options;
     store.removeAnnotations(store.savedAnnotations());
 
     // Cancel previously running search client.
     if (searchClient) {
       searchClient.cancel();
     }
 
-    if (uris.length > 0) {
-      searchAndLoad(uris, groupId);
-
+    if (uris && uris.length > 0) {
       streamFilter.resetFilter().addClause('/uri', 'one_of', uris);
       streamer.setConfig('filter', { filter: streamFilter.getFilter() });
     }
-  }
 
-  function searchAndLoad(uris, groupId) {
-    searchClient = new SearchClient(api.search, {
+    const searchOptions = {
       incremental: true,
+      maxResults: options.maxResults ?? 0,
```

Comment:
As above, I'd probably suggest to use `null` or `undefined` rather than `0` to indicate "don't put a limit on the number of results".